main.js

```diff
@@ -1884,27 +2053,57 @@ function getInstances() {
                 }
 
-                logger.debug('host.' + hostname + ' check instance "' + doc.rows[i].id  + '" for host "' + instance.common.host + '"');
-                console.log('host.' + hostname + ' check instance "' + doc.rows[i].id  + '" for host "' + instance.common.host + '"');
+                logger.debug(hostLogPrefix + ' check instance "' + doc.rows[i].id  + '" for host "' + instance.common.host + '"');
+                console.log(hostLogPrefix + ' check instance "' + doc.rows[i].id  + '" for host "' + instance.common.host + '"');
 
-                if (_ipArr.indexOf(instance.common.host) !== -1 || instance.common.host === hostname) {
-                    procs[instance._id] = procs[instance._id] || {};
-                    procs[instance._id].config = JSON.parse(JSON.stringify(instance));
-                    if (instance.common.enabled && (!instance.common.webExtension || !instance.native.webInstance)) count++;
-                }
+                if (checkAndAddInstance(instance, _ipArr) && instance.common.enabled && (!instance.common.webExtension || !instance.native.webInstance)) count++;
             }
 
             if (count > 0) {
-                logger.info('host.' + hostname + ' starting ' + count + ' instance' + (count > 1 ? 's' : ''));
+                logger.info(hostLogPrefix + ' starting ' + count + ' instance' + (count > 1 ? 's' : ''));
             } else {
-                logger.warn('host.' + hostname + ' does not start any instances on this host');
+                logger.warn(hostLogPrefix + ' does not start any instances on this host');
             }
         }
 
         initInstances();
     });
 }
 
+function instanceRelevantForThisHost(instance, _ipArr) {
+    if (!_ipArr.includes(instance.common.host) && instance.common.host !== hostname) return false;
+
+    // Normalize Compact group configuration
+    if (config.system.compact && instance.common.compact) {
+        if (instance.common.runAsCompactMode === undefined) instance.common.runAsCompactMode = true; // TODO repo logic!!
+        if (instance.common.compactGroup === undefined) instance.common.compactGroup = 0; // run in controller by default
+    }
+
+    if (compactGroupController) {
+        if (!config.system.compact || !instance.common.compact || !instance.common.runAsCompactMode) return false;
+        if (instance.common.runAsCompactMode && instance.common.compactGroup !== compactGroup) return false;
+    }
+    return true;
+}
+
+function checkAndAddInstance(instance, ipArr) {
+    if (!instanceRelevantForThisHost(instance, ipArr)) return false;
+    if (config.system.compact && instance.common.compact) {
+        if (instance.common.runAsCompactMode) {
+            compactProcs[instance.common.compactGroup] = compactProcs[instance.common.compactGroup] || {instances: []};
+        }
+    }
+
+    if (compactGroupController) {
+        logger.debug(hostLogPrefix + ' instance ' + instance._id + ' is relevant for this controller');
```

Comment:
How about `managed by` or `belongs to` instead of `relevant for`?

Reply:
managed by ...